waku/v2/waku_filter_v2/client.nim

```diff
@@ -77,25 +80,71 @@ proc ping*(wfc: WakuFilterClient, servicePeer: RemotePeerInfo): Future[FilterSub
 
   return await wfc.sendSubscribeRequest(servicePeer, filterSubscribeRequest)
 
-proc subscribe*(wfc: WakuFilterClient, servicePeer: RemotePeerInfo, pubsubTopic: PubsubTopic, contentTopics: seq[ContentTopic]): Future[FilterSubscribeResult] {.async.} =
-  let requestId = generateRequestId(wfc.rng)
-  let filterSubscribeRequest = FilterSubscribeRequest.subscribe(
-    requestId = requestId,
-    pubsubTopic = pubsubTopic,
-    contentTopics = contentTopics
-  )
-
-  return await wfc.sendSubscribeRequest(servicePeer, filterSubscribeRequest)
-
-proc unsubscribe*(wfc: WakuFilterClient, servicePeer: RemotePeerInfo, pubsubTopic: PubsubTopic, contentTopics: seq[ContentTopic]): Future[FilterSubscribeResult] {.async.} =
-  let requestId = generateRequestId(wfc.rng)
-  let filterSubscribeRequest = FilterSubscribeRequest.unsubscribe(
-    requestId = requestId,
-    pubsubTopic = pubsubTopic,
-    contentTopics = contentTopics
-  )
-
-  return await wfc.sendSubscribeRequest(servicePeer, filterSubscribeRequest)
+proc subscribe*(wfc: WakuFilterClient, servicePeer: RemotePeerInfo, pubsubTopic: Option[PubsubTopic], contentTopics: seq[ContentTopic]): Result[seq[Future[FilterSubscribeResult]], FilterSubscribeError] =
```

Comment:
I'm not sure I like this return type but would returning only Futures be better?